src/utils/date.js

```diff
@@ -3,12 +3,29 @@ import format from 'date-fns/format';
 import isToday from 'date-fns/is_today';
 import isYesterday from 'date-fns/is_yesterday';
 import isSameYear from 'date-fns/is_same_year';
+import addMinutes from 'date-fns/add_minutes';
 import tz from 'timezone/loaded';
 
 export { default as isSameDay } from 'date-fns/is_same_day';
 
+/**
+ * Use consistent timezone (UTC) in local and CI tests, for snapshot tests.
+ */
+// We're only counting on this if we can't set the timezone `Date` uses to
+// UTC in the first place; see note where we attempt to do that in
+// jest/globalSetup.js.
+function maybeAsUtc(date: Date) {
+  /* eslint-disable operator-linebreak */
+  /* eslint-disable-next-line id-match */
+  return __TEST__
+    ? // Negate the UTC offset, using an offset we get from `date` itself, so
+      // it'll be correct for DST-type changes through the year.
+      addMinutes(date, date.getTimezoneOffset())
```

Comment:
Have you confirmed empirically that this is the right way around? This is what someone suggested on an upstream issue thread, but it looked backward to me:
https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/Snapshot.20tests/near/1167551
but I could have a sign flipped.

Reply:
I did, but looks like I didn't write it down. At the tip of this revision (er, actually after the squashing you mentioned at https://github.com/zulip/zulip-mobile/pull/4670#discussion_r647965060; that shouldn't matter), I just did the following:

In `jest/globalSetup.js`:

```diff
-  process.env.TZ = 'UTC';
+  process.env.TZ = 'America/New_York';
```

Then I ran `tools/test jest`, and got only this failure (which I'd expect):

```
Summary of all failing tests
 FAIL  src/utils/__tests__/date-test.js
  ● Timezones (for test environment only) › Timezone has been set to UTC, unless on Windows

    expect(received).toBe(expected) // Object.is equality

    Expected: true
    Received: false

       8 |         // https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/Snapshot.20tests/near/1167560
       9 |         || process.platform === 'win32',
    > 10 |     ).toBe(true);
         |     ^
      11 |   });
      12 | });
      13 | 

      at Object.<anonymous> (src/utils/__tests__/date-test.js:10:5)


Test Suites: 1 failed, 55 passed, 56 total
Tests:       1 failed, 1 skipped, 1706 passed, 1708 total
Snapshots:   36 passed, 36 total
Time:        28.567 s, estimated 41 s
Ran all test suites related to changed files.
```

Then, I made this change in `maybePretendUtc` in `src/utils/date.js`:

```diff
-     addMinutes(date, date.getTimezoneOffset())
+     addMinutes(date, -date.getTimezoneOffset())
```

I ran `tools/test jest` again (er actually `tools/test jest > jest.txt 2>&1`), and I got that failure plus lots of others: 
[jest.txt](https://github.com/zulip/zulip-mobile/files/6627015/jest.txt)

Reply:
So be it! Thanks for confirming.